File: model/src/main/java/ro/upt/sma/heart/model/HeartMeasurementRepository.java
package ro.upt.sma.heart.model;

import java.util.List;

public interface HeartMeasurementRepository {

  interface HeartChangedListener {

    void onHeartChanged(HeartMeasurement heartMeasurement);
  }

  interface HeartListLoadedListener {

    void onHeartListLoaded(List<HeartMeasurement> heartMeasurements);
  }

  void post(HeartMeasurement heartMeasurement);

  void observe(HeartChangedListener listener);

  void retrieveAll(HeartListLoadedListener listener);

}

File: model/src/main/java/ro/upt/sma/heart/model/HeartMeasurementRepository.kt
package ro.upt.sma.heart.model

interface HeartMeasurementRepository {

    interface HeartChangedListener {
        fun onHeartChanged(heartMeasurement: HeartMeasurement)
    }

    interface HeartListLoadedListener {
        fun onHeartListLoaded(heartMeasurements: List<HeartMeasurement>)
    }

    fun post(heartMeasurement: HeartMeasurement)

    fun observe(listener: HeartChangedListener)

    fun retrieveAll(listener: HeartListLoadedListener)

}
